packaging/tags.py

```diff
@@ -0,0 +1,422 @@
+# This file is dual licensed under the terms of the Apache License, Version
+# 2.0, and the BSD License. See the LICENSE file in the root of this repository
+# for complete details.
+
+from __future__ import absolute_import
+
+import distutils.util
+import os
+
+try:
+    from os import fspath
+except ImportError:  # pragma: no cover
+    # From PEP 519.
+    def fspath(path):
+        """Return the string representation of the path.
+
+        If str or bytes is passed in, it is returned unchanged. If __fspath__()
+        returns something other than str or bytes then TypeError is raised. If
+        this function is given something that is not str, bytes, or os.PathLike
+        then TypeError is raised.
+        """
+        if isinstance(path, (str, bytes)):
+            return path
+
+        # Work from the object's type to match method resolution of other magic
+        # methods.
+        path_type = type(path)
+        try:
+            path = path_type.__fspath__(path)
+        except AttributeError:
+            if hasattr(path_type, "__fspath__"):
+                raise
+        else:
+            if isinstance(path, (str, bytes)):
+                return path
+            else:
+                raise TypeError(
+                    "expected __fspath__() to return str or bytes,"
+                    " not " + type(path).__name__
+                )
+
+        raise TypeError(
+            "expected str, bytes or os.PathLike object, not " + path_type.__name__
+        )
+
+
+import os.path
+import platform
+import re
+import sys
+import sysconfig
+import warnings
+
+
+INTERPRETER_SHORT_NAMES = {
+    "python": "py",  # Generic.
+    "cpython": "cp",
+    "pypy": "pp",
+    "ironpython": "ip",
+    "jython": "jy",
+}
+
+
+_32_BIT_INTERPRETER = sys.maxsize <= 2 ** 32
+
+
+# A dataclass would be better, but Python 2.7. :(
```

Comment:
No reason beyond it adds attrs as a dependency. But since you sound fine with that then I can bring it in and use it (just need to learn how to use the package 😄 ).